pkg/controller/daemon_controller.go

```diff
@@ -0,0 +1,432 @@
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/record"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/workqueue"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+	"github.com/golang/glog"
+)
+
+const (
+	// Daemon Controllers will periodically check that their daemons are running as expected.
+	FullDaemonControllerResyncPeriod = 30 * time.Second // TODO: Figure out if this time seems reasonable.
+	// Nodes don't need relisting.
+	FullNodeResyncPeriod = 0
+	// Daemon pods don't need relisting.
+	FullDaemonPodResyncPeriod = 0
+)
+
+type DaemonManager struct {
+	kubeClient client.Interface
+	podControl PodControlInterface
+
+	// To allow injection of syncDaemonController for testing.
+	syncHandler func(dcKey string) error
+	// A TTLCache of pod creates/deletes each dc expects to see
+	expectations ControllerExpectationsInterface
+	// A store of daemon controllers, populated by the podController.
+	dcStore cache.StoreToDaemonControllerLister
+	// A store of pods, populated by the podController
+	podStore cache.StoreToPodLister
+	// A store of pods, populated by the podController
+	nodeStore cache.StoreToNodeLister
+	// Watches changes to all pods.
+	dcController *framework.Controller
+	// Watches changes to all pods
+	podController *framework.Controller
+	// Watches changes to all nodes.
+	nodeController *framework.Controller
+	// Controllers that need to be updated.
+	queue *workqueue.Type
+}
+
+func NewDaemonManager(kubeClient client.Interface) *DaemonManager {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+
+	dm := &DaemonManager{
+		kubeClient: kubeClient,
+		podControl: RealPodControl{
+			kubeClient: kubeClient,
+			recorder:   eventBroadcaster.NewRecorder(api.EventSource{Component: "daemon-controller"}),
+		},
+		expectations: NewControllerExpectations(),
+		queue:        workqueue.New(),
+	}
+	// Manage addition/update of daemon controllers.
+	dm.dcStore.Store, dm.dcController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
+				return dm.kubeClient.DaemonControllers(api.NamespaceAll).List(labels.Everything())
+			},
+			WatchFunc: func(rv string) (watch.Interface, error) {
+				return dm.kubeClient.DaemonControllers(api.NamespaceAll).Watch(labels.Everything(), fields.Everything(), rv)
+			},
+		},
+		&api.DaemonController{},
+		FullDaemonControllerResyncPeriod,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: func(old interface{}) {
+				glog.Infoln("Adding dc")
```

Comment:
You need to run with -v=4 (or greater) to see the messages.

On Wed, Jul 8, 2015 at 7:02 PM, Ananya Kumar <notifications@github.com>
wrote:

> In pkg/controller/daemon_controller.go
> <https://github.com/vmarmol/kubernetes/pull/2#discussion_r34218075>:
>
> > +	}
> > +	// Manage addition/update of daemon controllers.
> > +	dm.dcStore.Store, dm.dcController = framework.NewInformer(
> > +		&cache.ListWatch{
> > +			ListFunc: func() (runtime.Object, error) {
> > +				return dm.kubeClient.DaemonControllers(api.NamespaceAll).List(labels.Everything())
> > +			},
> > +			WatchFunc: func(rv string) (watch.Interface, error) {
> > +				return dm.kubeClient.DaemonControllers(api.NamespaceAll).Watch(labels.Everything(), fields.Everything(), rv)
> > +			},
> > +		},
> > +		&api.DaemonController{},
> > +		FullDaemonControllerResyncPeriod,
> > +		framework.ResourceEventHandlerFuncs{
> > +			AddFunc: func(old interface{}) {
> > +				glog.Infoln("Adding dc")
>
> Changed to V(4), and printing the daemon's name.
> I'm keeping them because they help in debugging performance issues with
> the daemon.
> By the way, what do I need to do to enable V(4) logging?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vmarmol/kubernetes/pull/2/files#r34218075>.
>